### PR TITLE
feat: table interactions

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #2785 

### Description:

- [x] When you hover over a cell, the whole row should change background colour
- [x] When you click on a cell, it should appear to be selected. (border)
- [x] When you click on a cell, the context menu should appear (fixed to the bottom right of the cell)
- [x] Links should appear blue, and there should be an extra item in the context menu 'Go to link' (for now!)
- [x] You should still be able to easily copy the contents of the cell to your clipboard.

https://user-images.githubusercontent.com/962095/185170305-7ffddff4-a95f-45ab-a5c2-adad3a31f79d.mp4

### CODE REVIEW

File: packages/frontend/src/components/common/Table/ScrollableTable/RichBodyCell.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    :hover {
+        text-decoration: none;
+    }
+`;
+
+export const isUrl = (value: string) => {
+    return (
+        value &&
+        typeof value === 'string' &&
+        (value.startsWith('http://') || value.startsWith('https://'))
```
Comment (IrakliJani) [inline packages/frontend/src/components/common/Table/ScrollableTable/RichBodyCell.tsx:20]: I don't want to overengineer, but also this is not a good solution.

wdyt should we treat for example `goo.gl` as an URL without protocol

Comment (ZeRego) [reply packages/frontend/src/components/common/Table/ScrollableTable/RichBodyCell.tsx:20]: I think this approach is fine and force users to add a valid protocol.

File: packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx [hunk 2]
```diff
@@ -16,20 +19,28 @@ const TableBody = () => {
...
                             >
-                                <CellContextMenu cell={cell}>
-                                    {flexRender(
-                                        cell.column.columnDef.cell,
-                                        cell.getContext(),
-                                    )}
-                                </CellContextMenu>
-                            </BodyCell>
+                                <BodyCell
+                                    key={cell.id}
```
Comment (ZeRego) [inline packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx:28]: Should we move the `key` to the `CellContextMenu` ?

**Comment by ZeRego** [comment]
Found a funny interaction:
- make results table scroll
- select cell
- scroll with mouse wheel 
select and scroll

File: packages/frontend/src/components/Explorer/ResultsCard/CellContextMenu.tsx [hunk 3]
```diff
@@ -1,61 +1,120 @@
...
+    const value: ResultRow[0]['value'] = cell.getValue()?.value || {};
+
+    return (
+        <Menu>
+            {value.raw && isUrl(value.raw) && (
+                <MenuItem2
+                    icon="link"
+                    text="Go to link"
+                    onClick={() => {
+                        window.open(value.raw, '_blank');
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/components/Explorer/ResultsCard/CellContextMenu.tsx:48]: added

File: packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx]: good catch

**Comment by IrakliJani** [comment]
@ZeRego made changes to fix the scrolling bug. Popover has a backdrop when the menu is open but it prevents selecting content from the cell so I just add `overflow: hidden` to the table scroll container when the menu is open. Ideally, popover should hide when the "target component" is not visible anymore but could not find that in Blueprint docs

**Comment by IrakliJani** [comment]
@ZeRego ignore what I wrote above. I did not like the UX of the freezing table so I changed to this:

https://user-images.githubusercontent.com/962095/185387237-28f57ee9-7d41-4363-be71-e1120b73c581.mp4

**Review by ZeRego — APPROVED** [review]
lgtm 🚀

**Comment by owlas** [comment]
Usually there's a single row that happens to slightly overlap the totals row.

Screenshot 2022-08-18 at 15 19 20

**Comment by IrakliJani** [comment]
@owlas @ZeRego settled on freezing a table scrolling when the context menu is open.. that's how Notion does for example.

**Pull request merged**